### ISSUE TITLE
Squiz.PHP.EmbeddedPhp.ContentAfterEnd and DOCTYPE

### ISSUE THREAD

**Issue by kkmuffme** [squizlabs/PHP_CodeSniffer#3747]
**Describe the bug**

**Code sample**
```php
?><!DOCTYPE html>
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <description>If you are using a custom ruleset, please enter it here.</description>
  <rule ref="Squiz.PHP.EmbeddedPhp.ContentAfterEnd"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcbf test.php ...`
3. phpcbf created invalid HTML
```php
?>
<!DOCTYPE html>
```

**Expected behavior**
There is no space allowed before the DOCTYPE tag - any space before the DOCTYPE will cause browsers to run quirks mode.

**Versions (please complete the following information):**
 - OS: Centos 8
 - PHP: 7.4
 - PHPCS: master
 - Standard: custom

**Comment by rodrigoprimo** [comment]
I tried to reproduce this issue, but was not able to confirm the reported problem.

Using the following test file:

```php
<?php // First block. ?>
<?php
// Second block.
?><!DOCTYPE html>
<html>
<head><title>Test</title></head>
<body>
<p id="result"></p>
<script>
document.getElementById('result').textContent = 'Compat mode: ' + document.compatMode;
</script>
</body>
</html>
```

The sniff correctly reports the `ContentAfterEnd` error on line 4, and `phpcbf` fixes it to:

```php
<?php // First block. ?>
<?php
// Second block.
?>
<!DOCTYPE html>
<html>
<head><title>Test</title></head>
<body>
<p id="result"></p>
<script>
document.getElementById('result').textContent = 'Compat mode: ' + document.compatMode;
</script>
</body>
</html>
```

You can verify this by running both files through PHP and opening the resulting HTML files in a browser. Both will show `CSS1Compat` (standards mode), confirming that the fix does not trigger quirks mode. See https://developer.mozilla.org/en-US/docs/Web/API/Document/compatMode for more information about the different values of `compatMode`.

Did I miss anything? If not, I believe this issue can be closed.
